src/main/java/kitchenpos/products/tobe/domain/Product.java

```diff
@@ -0,0 +1,46 @@
+package kitchenpos.products.tobe.domain;
+
+import jakarta.persistence.*;
+
+import java.math.BigDecimal;
+import java.util.UUID;
+
+@Entity
+@Table(name = "product")
+public class Product {
+    @Column(name = "id", columnDefinition = "binary(16)")
+    @Id
+    private UUID id;
+
+    @Embedded
+    private DisplayedName name;
+
+    @Embedded
+    private Price price;
+
+    public Product() {}
+
+    public Product(UUID id, String name, Price price, ProductNameValidationService productNameValidationService) {
+        this.id = id;
+        this.name = new DisplayedName(name, productNameValidationService);
+        this.price = price;
+    }
+
+    public Product(UUID id, String name, BigDecimal price, ProductNameValidationService productNameValidationService) {
+        this.id = id;
+        this.name = new DisplayedName(name, productNameValidationService);
+        this.price = new Price(price);
+    }
```

Comment:
```suggestion
    public Product(
        UUID id,
        String name,
        BigDecimal price,
        ProductNameValidationService productNameValidationService
    ) {
        this(
            id,
            name,
            new Price(price),
            productNameValidationService
        );
    }
```

부 생성자는 주 생성자를 사용하는 것이 어떨까요?